instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractor.java

```diff
@@ -42,15 +41,29 @@ Optional<String> extractContainerId() {
       return empty();
     }
     try {
-      return filesystem
-          .lines(V2_CGROUP_PATH)
-          .filter(line -> line.contains("hostname"))
-          .flatMap(line -> Stream.of(line.split("/")))
-          .map(CONTAINER_ID_RE::matcher)
-          .filter(Matcher::matches)
-          .findFirst()
-          .map(matcher -> matcher.group(0));
-    } catch (IOException e) {
+      try (Stream<String> lines = filesystem.lines(V2_CGROUP_PATH)) {
```

Comment:
`filesystem.lines(..)` reads a file from disk. Would it make sense to read it once and keep the result in a list?

Reply:
It makes sense because the file is not big. I will change.

Reply:
you are still reading the file twice

Reply:
Sorry it is a bad mistake. Just fixed.